### PR TITLE
(FMT): fix #451

### LINKED ISSUE #451: Properly indent known macro calls
```Rust
fn foo() {
    write!(&mut self.destination, "{}",
           magic_number);
}
```

gets reformatted to

```Rust
fn foo() {
    write!(&mut self.destination, "{}",
    magic_number);
}
```

And it really should not. It is important that `write!` is a known format-like macro. Unknown macros are properly left intact. @jajakobyly you might want to look into this :)

### PR DESCRIPTION
**Pull request by jajakobyly**

### CODE REVIEW

File: src/test/kotlin/org/rust/ide/formatter/RustFormatterTestCase.kt [hunk 1]
```diff
@@ -33,4 +33,6 @@ class RustFormatterTestCase : FormatterTestCase() {
     // FIXME: these two guys are way too big
     fun testSpacing() = doTest()
     fun testLineBreaks() = doTest()
+
+    fun testIssue451() = doTest()   // https://github.com/intellij-rust/intellij-rust/issues/451
```
Comment (matklad) [inline src/test/kotlin/org/rust/ide/formatter/RustFormatterTestCase.kt]: Pretty scary name for the issue, I like it! :)

**Comment by jajakobyly** [comment]
I'm not sure whether the behaviour for `write!{...}` is correct. rustfmt seems to not touch it

**Comment by matklad** [comment]
I think write should be handled the same way as a normal function call (+ some handling of keyword arguments)

**Comment by matklad** [comment]
Thanks!

**Comment by matklad** [comment]
FYI: rustraytracer now is fully reformatted with `ctrl+alt+l` :)

**Pull request merged**